raiden_contracts/tests/test_channel_cooperative_settle.py

```diff
@@ -439,49 +562,73 @@ def test_cooperative_close_replay_reopened_channel(
     assert channel_identifier1 != channel_identifier2
     with pytest.raises(TransactionFailed):
         token_network.functions.cooperativeSettle(
-            channel_identifier2, B, balance_B, A, balance_A, signature_B, signature_A
+            channel_identifier2,
+            (B, balance_B, expiration, signature_B2, signature_A2),
+            (A, balance_A, expiration, signature_A1, signature_B1),
         ).call({"from": B})
 
     # Signed message with the correct channel identifier must work
-    (signature_A2, signature_B2) = create_cooperative_settle_signatures(
-        [A, B], channel_identifier2, B, balance_B, A, balance_A
+    (signature_A1, signature_B1) = create_withdraw_signatures(
+        [A, B], channel_identifier2, A, balance_A, expiration
+    )
+    (signature_A2, signature_B2) = create_withdraw_signatures(
+        [A, B], channel_identifier2, B, balance_B, expiration
     )
     call_and_transact(
         token_network.functions.cooperativeSettle(
-            channel_identifier2, B, balance_B, A, balance_A, signature_B2, signature_A2
+            channel_identifier2,
+            (B, balance_B, expiration, signature_B2, signature_A2),
+            (A, balance_A, expiration, signature_A1, signature_B1),
         ),
         {"from": B},
     )
 
 
-@pytest.mark.skip(reason="Delayed until another milestone")
 def test_cooperative_settle_channel_event(
     get_accounts: Callable,
     token_network: Contract,
     create_channel: Callable,
     channel_deposit: Callable,
-    create_cooperative_settle_signatures: Callable,
+    create_withdraw_signatures: Callable,
     event_handler: Callable,
 ) -> None:
     ev_handler = event_handler(token_network)
     (A, B) = get_accounts(2)
     deposit_A = 10
     balance_A = 2
     balance_B = 8
+    expiration = 100
+
     channel_identifier = create_channel(A, B)[0]
     channel_deposit(channel_identifier, A, deposit_A, B)
 
-    (signature_A, signature_B) = create_cooperative_settle_signatures(
-        [A, B], channel_identifier, B, balance_B, A, balance_A
+    (signature_A1, signature_B1) = create_withdraw_signatures(
+        [A, B], channel_identifier, A, balance_A, expiration
+    )
+    (signature_A2, signature_B2) = create_withdraw_signatures(
+        [A, B], channel_identifier, B, balance_B, expiration
     )
 
     txn_hash = call_and_transact(
         token_network.functions.cooperativeSettle(
-            channel_identifier, B, balance_B, A, balance_A, signature_B, signature_A
+            channel_identifier,
+            (B, balance_B, expiration, signature_B2, signature_A2),
+            (A, balance_A, expiration, signature_A1, signature_B1),
         ),
         {"from": B},
     )
 
+    ev_handler.add(
+        txn_hash, ChannelEvent.WITHDRAW, check_withdraw(channel_identifier, B, balance_B)
+    )
+    # ev_handler.add(
+    #     txn_hash,
+    #     ChannelEvent.WITHDRAW,
+    #     check_withdraw(channel_identifier, A, balance_A)
+    # )
```

Comment:
Why is this commented out? It looks like it should work.

Reply:
The event checking code cannot handle checking two events with the same name in the same tx right now. I didn't want to implement that before a first review on the contract changes.